packages/ember-data/lib/serializers/json-api-serializer.js

```diff
@@ -0,0 +1,299 @@
+/**
+  @module ember-data
+*/
+
+import JSONSerializer from 'ember-data/serializers/json-serializer';
+import normalizeModelName from 'ember-data/system/normalize-model-name';
+import { pluralize, singularize } from 'ember-inflector/lib/system/string';
+
+var dasherize = Ember.String.dasherize;
+
+/**
+  @class JSONAPISerializer
+  @namespace DS
+  @extends DS.JSONSerializer
+*/
+export default JSONSerializer.extend({
+
+  /**
+    @method _normalizeResponse
+    @param {DS.Store} store
+    @param {DS.Model} primaryModelClass
+    @param {Object} payload
+    @param {String|Number} id
+    @param {String} requestType
+    @param {Boolean} isSingle
+    @return {Object} JSON-API Document
+    @private
+  */
+  _normalizeResponse: function(store, primaryModelClass, payload, id, requestType, isSingle) {
+
+    let normalizeResourceHelper = (resourceHash) => {
```

Comment:
we should probably move this to the prototype to avoid re-creating a closure every time.